### PR TITLE
Require Python 3.6 or newer, use f-strings and variable type annotation syntax

### LINKED ISSUE #314: Drop support for Python 3.5
Until Buster is released (planned for July 2019, <https://lists.debian.org/debian-devel-announce/2019/06/msg00003.html>), it's still the newest Python in Debian, so it would be nice to keep for now. (PyPy has 3.6 beta since 7.1.0, I don't know which version Travis-CI has.)

Known useful new features:
* f-strings
* variable annotation syntax

### PR DESCRIPTION
**Pull request by mmaslowskicc**
Fixes #314

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,6 @@ dist: xenial
 language: python
 conditions: v1
 python:
-- 3.5
 - 3.6
 - 3.7
```
Comment (fizyk) [inline .travis.yml]: could you try if there's 3.8 available already?

Comment (mmaslowskicc) [reply .travis.yml]: It's available, while a test fails due to the fix for <https://pagure.io/python-daemon/issue/34> not being released. Should I add 3.8 and xfail the affected tests on that version?

Comment (fizyk) [reply .travis.yml]: use this for now: https://docs.travis-ci.com/user/customizing-the-build/#rows-that-are-allowed-to-fail

Comment (fizyk) [reply .travis.yml]: or the xfail indeed with failing test while it starts passing.

**Pull request merged**